src/modules/emote_autocomplete/components/Emotes.jsx

```diff
@@ -0,0 +1,138 @@
+import React, {useEffect, useRef, useState, useMemo, useCallback} from 'react';
+import keyCodes from '../../../utils/keycodes.js';
+import emoteMenuViewStore from '../../../common/stores/emote-menu-view-store.js';
+import useChatInputPartialEmote from '../hooks/ChatInputPartialEmote.jsx';
+import EmoteRow from './EmoteRow.jsx';
+import EmotesHeader from './EmotesHeader.jsx';
+import styles from './Emotes.module.css';
+import useEmoteMenuViewStoreUpdated from '../../../common/hooks/EmoteMenuViewStore.jsx';
+
+const MAX_EMOTES_SHOWN = 8;
+
+function travelUp(currentSelection, rowCount) {
+  const newSelection = currentSelection - 1;
+
+  if (newSelection <= -1) {
+    return rowCount - 1;
+  }
+
+  return newSelection;
+}
+
+function travelDown(currentSelection, rowCount) {
+  const newSelection = currentSelection + 1;
+
+  if (newSelection > rowCount - 1) {
+    return 0;
+  }
+
+  return newSelection;
+}
+
+export default function Emotes({chatInputElement, repositionPopover, onComplete, getChatInputPartialEmote}) {
+  const chatInputPartialEmote = useChatInputPartialEmote(chatInputElement, getChatInputPartialEmote);
+  const [matches, setMatches] = useState([]);
+  const shortMatches = useMemo(() => matches.slice(0, MAX_EMOTES_SHOWN), [matches]);
+
+  const computeMatches = useCallback(() => {
+    const searchedEmotes = emoteMenuViewStore.search(chatInputPartialEmote);
+    setMatches(searchedEmotes.map(({item}) => item));
+  }, [chatInputPartialEmote]);
+
+  useEmoteMenuViewStoreUpdated(false, computeMatches);
```

Comment:
why would this always be false? isn't the point of this hook to listen to store updates and update the store as needed?